File: shareapp/app/src/main/java/com/example/shareapp/MainActivity.java
package com.example.shareapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button sharebtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        sharebtn = findViewById(R.id.sharebtn);
        sharebtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent shareintent = new Intent(Intent.ACTION_SEND);
                shareintent.setType("text/plain");
                String shareBody = "Download this one :  https://play.google.com/store/apps/details?id=com.hwkrbbt.downloadall&hl=en_IN&gl=US";
                String sharesubject = "learning the share app";

                shareintent.putExtra(Intent.EXTRA_SUBJECT,sharesubject);
                shareintent.putExtra(Intent.EXTRA_TEXT, shareBody);
                startActivity(Intent.createChooser(shareintent,"Share Using"));

                //startActivity(shareintent);
            }
        });
    }
}

File: opendialertest/app/src/main/java/com/example/opendialertest/MainActivity.kt
package com.example.opendialertest

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var btn= findViewById<Button>(R.id.button)
        btn.setOnClickListener(View.OnClickListener {
            var intent = Intent(Intent.ACTION_DIAL, Uri.parse("tel:1234567890"))
            startActivity(intent)
        })
    }
}